portfolio/src/main/java/com/google/sps/data/AUX.java

```diff
@@ -35,22 +36,57 @@ private static void GetDevKey(){
         }
     }
 
-    public static Video VideoIdToObject(String videoId) throws IOException, GeneralSecurityException{
+    public static Video VideoIdToObject(String videoId){
         GetDevKey();
-        YouTube youtubeService = getService();
-        // Define and execute the API request
-        YouTube.Videos.List videoRequest = youtubeService.videos()
+
+        VideoListResponse videoResponse = new VideoListResponse();
+        try{
+            //Get youtube service
+            YouTube youtubeService = getService();
+
+            // Define and execute the API request
+            YouTube.Videos.List videoRequest = youtubeService.videos()
             .list("snippet");
-        VideoListResponse videoResponse = videoRequest.setKey(DEVELOPER_KEY)
+
+            videoResponse = videoRequest.setKey(DEVELOPER_KEY)
             .setId(videoId)
             .execute();
-        com.google.api.services.youtube.model.Video video = videoResponse.getItems().get(0);
-        String videoTitle = video.getSnippet().getTitle();
-        String videoDescription = video.getSnippet().getLocalized().getDescription();
-        String thumbnailUrl = video.getSnippet().getThumbnails().getDefault().getUrl();
-        boolean isPublic = true;
-        Video result = new Video(videoId, videoTitle, videoDescription, thumbnailUrl, isPublic);
-        return result;
+
+            com.google.api.services.youtube.model.Video video = videoResponse.getItems().get(0);
+            String videoTitle = video.getSnippet().getTitle();
+            String videoDescription = video.getSnippet().getLocalized().getDescription();
+            String thumbnailUrl = video.getSnippet().getThumbnails().getDefault().getUrl();
+            boolean isPublic = true;
+            Video result = new Video(videoId, videoTitle, videoDescription, thumbnailUrl, isPublic);
```

Comment:
Can leave this as 
new Video(videoId, videoTitle, videoDescription, thumbnailUrl, /* isPublic= */ true).